server/src/test/java/io/unitycatalog/server/utils/PopulateTestDatabase.java

```diff
@@ -278,59 +279,53 @@ public static void main(String[] args) throws JsonProcessingException {
-    ColumnInfoDAO age =
-        ColumnInfoDAO.builder()
-            .name("age")
-            .typeName(ColumnTypeName.LONG.getValue())
-            .comment("long column")
-            .ordinalPosition((short) 1)
-            .build();
-    addTypeTextAndJsonText(age);
-    ColumnInfoDAO country =
-        ColumnInfoDAO.builder()
-            .name("country")
-            .typeName(ColumnTypeName.STRUCT.getValue())
-            .comment("partition column")
-            .ordinalPosition((short) 2)
-            .partitionIndex((short) 0)
-            .build();
-    addTypeTextAndJsonText(country);
+
     String partitionedTableName = "user_countries";
     String partitionedStorageRoot = "etc/data/external/";
     String partitionedTablePath = partitionedStorageRoot + catalogName + "/" + schemaName + "/tables/" + partitionedTableName;
-    UUID partitionedTableId = UUID.randomUUID();
-
-    TableInfoDAO partitionedTableInfoDAO =
-        TableInfoDAO.builder()
-            .id(partitionedTableId)
-            .name(partitionedTableName)
-            .schemaId(UUID.fromString(schemaId))
-            .comment("Partitioned table")
-            .columns(List.of(firstName, age, country))
-            .dataSourceFormat(DataSourceFormat.DELTA.getValue())
-            .type(TableType.EXTERNAL.getValue())
-            .createdAt(new Date())
-            .updatedAt(new Date())
-            .url(partitionedTablePath)
-            .build();
-    partitionedTableInfoDAO
-        .getColumns()
-        .forEach(columnInfoDAO -> columnInfoDAO.setTable(partitionedTableInfoDAO));
-
-    try (Session session = factory.openSession()) {
-      Transaction tx = session.beginTransaction();
-      session.persist(partitionedTableInfoDAO);
-      tx.commit();
-    }
 
+    TableRepository tableRepository = TableRepository.getInstance();
+
+    CreateTable createExternalPartitionedTable = new CreateTable();
+    createExternalPartitionedTable.setName("user_countries");
+    createExternalPartitionedTable.setComment("Partitioned table");
+    createExternalPartitionedTable.setTableType(TableType.EXTERNAL);
+    createExternalPartitionedTable.setDataSourceFormat(DataSourceFormat.DELTA);
+    createExternalPartitionedTable.setStorageLocation(partitionedTablePath);
+    createExternalPartitionedTable.setCatalogName("unity");
+    createExternalPartitionedTable.setSchemaName("default");
+
+    // Add columns
+    ColumnInfo firstName = new ColumnInfo();
+    firstName.setName("first_name");
+    firstName.setTypeName(ColumnTypeName.STRING);
+    firstName.setTypeText(ColumnTypeName.STRING.getValue());
+    firstName.setTypeJson(ColumnTypeName.STRUCT.getValue());
+    firstName.setComment("string column");
+    firstName.setPosition(0);
+    createExternalPartitionedTable.addColumnsItem(firstName);
+
+    ColumnInfo age = new ColumnInfo();
+    age.setName("age");
+    age.setTypeName(ColumnTypeName.LONG);
+    age.setTypeText(ColumnTypeName.LONG.getValue());
```

Comment:
Setting these generic values might pose problems in a third party client which is trying to interpret these columns (We observed the same with duckdb). Make sure you set these values keeping in mind the consumers. This goes for both `typeText` and `typeJson`.

Reply:
Without setting these values (`typeName`, `typeText`) I was getting an NPE from [this code block](https://github.com/unitycatalog/unitycatalog/blob/209895f0409a8bf87f6a7fa0de30d1868d6d6727/server/src/main/java/io/unitycatalog/server/persist/TableRepository.java#L109)...

Reply:
I just meant, make sure your final client understands these values, whatever being set here.

Reply:
@ravivj-db Thanks for mentioning it! I made it work with DuckDB by modifying how I pass the JSON value and also by adding required by DuckDB precision and scale.